lighthouse-core/gather/gather-runner.js

```diff
@@ -165,6 +166,20 @@ class GatherRunner {
     } else if (mainRecord.hasErrorStatusCode()) {
       errorDef = {...LHError.errors.ERRORED_DOCUMENT_REQUEST};
       errorDef.message += ` Status code: ${mainRecord.statusCode}.`;
+    } else if (!mainRecord.finished) {
+      // could be security error
+      const securityState = await driver.getSecurityState();
+      if (securityState.securityState === 'insecure') {
+        errorDef = {...LHError.errors.INSECURE_DOCUMENT_REQUEST};
+        const insecureDescriptions = securityState.explanations
+          .filter(exp => exp.securityState === 'insecure')
+          .map(exp => exp.description)
+          .join(' ');
+        errorDef.message += ` Insecure: ${insecureDescriptions}`;
```

Comment:
Agree the explanation gathering is the best we can do.
For this line i think it could just be 

```suggestion
        errorDef.message += ` ${insecureDescriptions.join(' ')}`;
```